assets/js/campaign-editors/markdown/menu.js

```diff
@@ -0,0 +1,272 @@
+import { toggleMark } from "prosemirror-commands"
+import { liftListItem, sinkListItem, wrapInList } from "prosemirror-schema-list"
+import { NodeSelection, Plugin, Selection, TextSelection } from "prosemirror-state"
+import { canInsert, hasActiveMark, hasActiveNodeType, toggleBlockType } from "./helpers"
+import { markdownParser } from "./markdown-parser"
+import { schema } from "./markdown-schema"
+import { MenuButton } from "./menu-button"
+
+/** Class for handling editor menus. */
+class MenuView {
+  /**
+   * Initializes a new editor menu.
+   * @param {MenuButton[]} items - MenuButtons
+   * @param {Object} editorView - ProseMirror EditorView
+   */
+  constructor(items, editorView) {
+    this.items = items
+    this.editorView = editorView
+    this.dom = editorView.dom.parentNode.parentNode.querySelector(".wysiwyg--menu")
+
+    items.forEach((item) => item.attach(editorView.dom.parentNode.parentNode))
+
+    this.update()
+
+    this.dom.addEventListener("click", e => {
+      e.preventDefault()
+    })
+
+    this.dom.addEventListener("mousedown", e => {
+      e.stopPropagation()
+      e.preventDefault()
+      editorView.focus()
+      const activatedItem = items.find(({ dom }) => dom.contains(e.target))
+      if (activatedItem) activatedItem.exec(editorView)
+    }, true)
+  }
+
+  update() {
+    this.items.forEach(item => item.update(this.editorView.state))
```

Comment:
The editorView.state will always be available?

Reply:
Yep, it comes from Prosemirror.